src/features/defi/providers/idle/components/IdleManager/Claim/components/Status.tsx

```diff
@@ -0,0 +1,219 @@
+  selectAssetById,
+  selectFirstAccountSpecifierByChainId,
+  selectMarketDataById,
+  selectTxById,
+} from 'state/slices/selectors'
+import { serializeTxIndex } from 'state/slices/txHistorySlice/utils'
+import { useAppSelector } from 'state/store'
+
+import { IdleClaimActionType } from '../ClaimCommon'
+import { ClaimContext } from '../ClaimContext'
+
+export const Status = () => {
+  const translate = useTranslate()
+  const { state, dispatch } = useContext(ClaimContext)
+  const { query, history: browserHistory } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, contractAddress: vaultAddress, assetReference } = query
+
+  const assetNamespace = 'erc20'
+  // Asset info
+  const underlyingAssetId = toAssetId({
+    chainId,
+    assetNamespace,
+    assetReference,
+  })
+  const underlyingAsset = useAppSelector(state => selectAssetById(state, underlyingAssetId))
+  const assetId = toAssetId({
+    chainId,
+    assetNamespace,
+    assetReference: vaultAddress,
+  })
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const feeAssetId = toAssetId({
+    chainId,
+    assetNamespace: 'slip44',
+    assetReference: ASSET_REFERENCE.Ethereum,
+  })
+  const feeAsset = useAppSelector(state => selectAssetById(state, feeAssetId))
+  const feeMarketData = useAppSelector(state => selectMarketDataById(state, feeAssetId))
+
+  const accountSpecifier = useAppSelector(state =>
+    selectFirstAccountSpecifierByChainId(state, chainId),
+  )
+
+  const serializedTxIndex = useMemo(() => {
+    if (!(state?.txid && state?.userAddress)) return ''
+    return serializeTxIndex(accountSpecifier, state.txid, state.userAddress)
+  }, [state?.txid, state?.userAddress, accountSpecifier])
+  const confirmedTransaction = useAppSelector(gs => selectTxById(gs, serializedTxIndex))
+
+  useEffect(() => {
+    if (confirmedTransaction && confirmedTransaction.status !== 'Pending' && dispatch) {
+      dispatch({
+        type: IdleClaimActionType.SET_CLAIM,
+        payload: {
+          txStatus: confirmedTransaction.status === 'Confirmed' ? 'success' : 'failed',
+          usedGasFee: confirmedTransaction.fee?.value,
+        },
+      })
+    }
+  }, [confirmedTransaction, dispatch])
+
+  let renderAssets: any[] = []
+  useAppSelector(selectorState => {
+    if (state && state.claimableTokens) {
+      state.claimableTokens.forEach(token => {
+        const asset = selectAssetById(selectorState, token.assetId)
+        if (asset) {
+          renderAssets.push(
+            <Stack direction='row' alignItems='center' justifyContent='center' key={token.assetId}>
+              <AssetIcon boxSize='8' src={asset.icon} />
+              <Amount.Crypto
+                fontSize='lg'
+                fontWeight='medium'
+                value={bnOrZero(token.amount).div(`1e+${asset.precision}`).toString()}
+                symbol={asset?.symbol}
+              />
+            </Stack>,
+          )
+        }
+      })
+    }
+  })
```

Comment:
really don't like this let and mutate patter, and especially don't like logic within the selector function